Mathlib/NumberTheory/LSeries/Deriv.lean

```diff
@@ -0,0 +1,227 @@
+lemma LSeries.deriv {f : ℕ → ℂ} {s : ℂ} (h : abscissaOfAbsConv f < s.re) :
+    deriv (LSeries f) s = - LSeries (logMul f) s :=
+  (hasDerivAt h).deriv
+
+/-- The derivative of the L-series of `f` agrees with the negative of the L-series of
+`log * f` on the right half-plane of absolute convergence. -/
+protected
+lemma LSeries.deriv_eqOn {f : ℕ → ℂ} :
+    {s | abscissaOfAbsConv f < s.re}.EqOn (deriv (LSeries f)) (- LSeries (logMul f)) :=
+  deriv_eqOn (isOpen_rightHalfPlane _) fun _ hs ↦ (hasDerivAt hs).hasDerivWithinAt
+
+/-- If the L-series of `f` is summable at `s` and `re s < re s'`, then the L-series of the
+point-wise product of `log` with `f` is summable at `s'`. -/
+lemma LSeriesSummable.logMul_of_lt_re {f : ℕ → ℂ} {s : ℂ} (h : abscissaOfAbsConv f < s.re) :
+    LSeriesSummable (logMul f) s :=
+  (LSeriesSummable_logMul_and_hasDerivAt h).1
+
+/-- If the L-series of the point-wise product of `log` with `f` is summable at `s`, then
+so is the L-series of `f`. -/
+lemma LSeriesSummable.of_logMul {f : ℕ → ℂ} {s : ℂ} (h : LSeriesSummable (logMul f) s) :
+    LSeriesSummable f s := by
+  refine h.norm.of_norm_bounded_eventually_nat (fun n ↦ ‖term (logMul f) s n‖) ?_
+  simp only [norm_div, natCast_log, norm_mul, Filter.eventually_atTop]
+  -- We use that `1 ≤ log n` for `n ≥ 3`.
+  refine ⟨3, fun n hn ↦ ?_⟩
+  simp only [term_of_ne_zero (show n ≠ 0 by omega), logMul, norm_mul, mul_div_assoc]
+  refine le_mul_of_one_le_left (norm_nonneg _) ?_
+  rw [← natCast_log, norm_eq_abs, abs_ofReal, ← Real.log_exp 1,
+    _root_.abs_of_nonneg <| Real.log_nonneg <| by norm_cast; linarith]
+  exact Real.log_le_log (Real.exp_pos 1) <| (Real.exp_one_lt_d9.trans <| by norm_num).le.trans <|
+    (show (3 : ℝ) ≤ n by exact_mod_cast hn)
+
+/-- The abscissa of absolute convergence of the point-wise product of `log` and `f`
+is the same as that of `f`. -/
+@[simp]
+lemma abscissaOfAbsConv_logMul {f : ℕ → ℂ} :
+    abscissaOfAbsConv (logMul f) = abscissaOfAbsConv f :=
+  le_antisymm (abscissaOfAbsConv_le_of_forall_lt_LSeriesSummable'
+      fun y hy ↦ LSeriesSummable.logMul_of_lt_re <| by simp only [ofReal_re, hy]) <|
+    abscissaOfAbsConv_le_of_forall_lt_LSeriesSummable' fun y hy ↦
+      (LSeriesSummable_of_abscissaOfAbsConv_lt_re <| by simp only [ofReal_re, hy]).of_logMul
+
+/-!
+### Higher derivatives of L-series
+-/
+
+/-- The higher derivatives of the terms of an L-series. -/
+lemma LSeries.iteratedDeriv_term (f : ℕ → ℂ) (m n : ℕ) (s : ℂ) :
+    iteratedDeriv m (fun z ↦ term f z n) s =
+      (-1) ^ m * (term (logPowMul m f) s n) := by
+  induction' m with m ih generalizing f s
+  · simp only [Nat.zero_eq, iteratedDeriv_zero, pow_zero, logPowMul_zero, one_mul]
+  · have ih' : iteratedDeriv m (fun s ↦ term (logMul f) s n) =
+        fun s ↦ (-1) ^ m * (term (logPowMul m (logMul f)) s n) :=
+      funext <| ih (logMul f)
+    rw [iteratedDeriv_succ', deriv_term' f n, iteratedDeriv_neg, ih', neg_mul_eq_neg_mul,
+      logPowMul_succ', pow_succ, neg_one_mul]
```

Comment:
Do you actually use this? It seems it is not used in `LSeries.iteratedDeriv`, so I suggest deleting it, it's not particularly interesting in itself.

Reply:
OK.